### ISSUE TITLE
Python 3.11 produces an error when using the __repr__ method on Events

### ISSUE THREAD

**Issue by lewisoptellum** [pytorch/ignite#2997]
## 🐛 Bug description

When using the `__repr__` method using py3.11 an event produces `AttributeError: 'str' object has no attribute 'name'` . This doesn't happen using py3.8. Perhaps related to #2904 

This can be reproduced:

 ```python
  completed = ignite.engine.Events.COMPLETED
  print(completed) #works
  print(f"{completed!r}") #doesn't work 
```

## Environment

- PyTorch Version (e.g., 1.4): 2.0.1
- Ignite Version (e.g., 0.3.0): 0.4.12
- OS (e.g., Linux): Ubuntu 22.04.2 LTS
- How you installed Ignite (`conda`, `pip`, source): pip
- Python version: 3.11.3

```[tasklist]
### Tasks
```

**Comment by sadra-barikbin** [comment]
@lewisoptellum , thank you for reporting that. This is originally related to a bug in Python3.11's enum and is fixed as of `3.11.4` (I checked that).

**Comment by vfdev-5** [comment]
Closing as resolved. @lewisoptellum if you still experience this problem please reopen the issue.
